// @generated/type-graphql/resolvers/relations/CourseQuiz/CourseQuizRelationsResolver.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CourseQuizRelationsResolver = void 0;
const tslib_1 = require("tslib");
const TypeGraphQL = (0, tslib_1.__importStar)(require("type-graphql"));
const CourseBatches_1 = require("../../../models/CourseBatches");
const CourseQuiz_1 = require("../../../models/CourseQuiz");
const Courses_1 = require("../../../models/Courses");
const helpers_1 = require("../../../helpers");
let CourseQuizRelationsResolver = class CourseQuizRelationsResolver {
    async courseBatches(courseQuiz, ctx) {
        return (0, helpers_1.getPrismaFromContext)(ctx).courseQuiz.findUnique({
            where: {
                id: courseQuiz.id,
            },
        }).courseBatches({});
    }
    async courses(courseQuiz, ctx) {
        return (0, helpers_1.getPrismaFromContext)(ctx).courseQuiz.findUnique({
            where: {
                id: courseQuiz.id,
            },
        }).courses({});
    }
};
(0, tslib_1.__decorate)([
    TypeGraphQL.FieldResolver(_type => CourseBatches_1.CourseBatches, {
        nullable: false
    }),
    (0, tslib_1.__param)(0, TypeGraphQL.Root()),
    (0, tslib_1.__param)(1, TypeGraphQL.Ctx()),
    (0, tslib_1.__metadata)("design:type", Function),
    (0, tslib_1.__metadata)("design:paramtypes", [CourseQuiz_1.CourseQuiz, Object]),
    (0, tslib_1.__metadata)("design:returntype", Promise)
], CourseQuizRelationsResolver.prototype, "courseBatches", null);
(0, tslib_1.__decorate)([
    TypeGraphQL.FieldResolver(_type => Courses_1.Courses, {
        nullable: false
    }),
    (0, tslib_1.__param)(0, TypeGraphQL.Root()),
    (0, tslib_1.__param)(1, TypeGraphQL.Ctx()),
    (0, tslib_1.__metadata)("design:type", Function),
    (0, tslib_1.__metadata)("design:paramtypes", [CourseQuiz_1.CourseQuiz, Object]),
    (0, tslib_1.__metadata)("design:returntype", Promise)
], CourseQuizRelationsResolver.prototype, "courses", null);
CourseQuizRelationsResolver = (0, tslib_1.__decorate)([
    TypeGraphQL.Resolver(_of => CourseQuiz_1.CourseQuiz)
], CourseQuizRelationsResolver);
exports.CourseQuizRelationsResolver = CourseQuizRelationsResolver;
