// @generated/type-graphql/resolvers/crud/Faq/args/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpsertFaqArgs = exports.UpdateManyFaqArgs = exports.UpdateFaqArgs = exports.GroupByFaqArgs = exports.FindUniqueFaqArgs = exports.FindManyFaqArgs = exports.FindFirstFaqArgs = exports.DeleteManyFaqArgs = exports.DeleteFaqArgs = exports.CreateManyFaqArgs = exports.CreateFaqArgs = exports.AggregateFaqArgs = void 0;
var AggregateFaqArgs_1 = require("./AggregateFaqArgs");
Object.defineProperty(exports, "AggregateFaqArgs", { enumerable: true, get: function () { return AggregateFaqArgs_1.AggregateFaqArgs; } });
var CreateFaqArgs_1 = require("./CreateFaqArgs");
Object.defineProperty(exports, "CreateFaqArgs", { enumerable: true, get: function () { return CreateFaqArgs_1.CreateFaqArgs; } });
var CreateManyFaqArgs_1 = require("./CreateManyFaqArgs");
Object.defineProperty(exports, "CreateManyFaqArgs", { enumerable: true, get: function () { return CreateManyFaqArgs_1.CreateManyFaqArgs; } });
var DeleteFaqArgs_1 = require("./DeleteFaqArgs");
Object.defineProperty(exports, "DeleteFaqArgs", { enumerable: true, get: function () { return DeleteFaqArgs_1.DeleteFaqArgs; } });
var DeleteManyFaqArgs_1 = require("./DeleteManyFaqArgs");
Object.defineProperty(exports, "DeleteManyFaqArgs", { enumerable: true, get: function () { return DeleteManyFaqArgs_1.DeleteManyFaqArgs; } });
var FindFirstFaqArgs_1 = require("./FindFirstFaqArgs");
Object.defineProperty(exports, "FindFirstFaqArgs", { enumerable: true, get: function () { return FindFirstFaqArgs_1.FindFirstFaqArgs; } });
var FindManyFaqArgs_1 = require("./FindManyFaqArgs");
Object.defineProperty(exports, "FindManyFaqArgs", { enumerable: true, get: function () { return FindManyFaqArgs_1.FindManyFaqArgs; } });
var FindUniqueFaqArgs_1 = require("./FindUniqueFaqArgs");
Object.defineProperty(exports, "FindUniqueFaqArgs", { enumerable: true, get: function () { return FindUniqueFaqArgs_1.FindUniqueFaqArgs; } });
var GroupByFaqArgs_1 = require("./GroupByFaqArgs");
Object.defineProperty(exports, "GroupByFaqArgs", { enumerable: true, get: function () { return GroupByFaqArgs_1.GroupByFaqArgs; } });
var UpdateFaqArgs_1 = require("./UpdateFaqArgs");
Object.defineProperty(exports, "UpdateFaqArgs", { enumerable: true, get: function () { return UpdateFaqArgs_1.UpdateFaqArgs; } });
var UpdateManyFaqArgs_1 = require("./UpdateManyFaqArgs");
Object.defineProperty(exports, "UpdateManyFaqArgs", { enumerable: true, get: function () { return UpdateManyFaqArgs_1.UpdateManyFaqArgs; } });
var UpsertFaqArgs_1 = require("./UpsertFaqArgs");
Object.defineProperty(exports, "UpsertFaqArgs", { enumerable: true, get: function () { return UpsertFaqArgs_1.UpsertFaqArgs; } });
