// @generated/type-graphql/resolvers/relations/CourseAssignment/CourseAssignmentRelationsResolver.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CourseAssignmentRelationsResolver = void 0;
const tslib_1 = require("tslib");
const TypeGraphQL = (0, tslib_1.__importStar)(require("type-graphql"));
const CourseAssignment_1 = require("../../../models/CourseAssignment");
const CourseBatches_1 = require("../../../models/CourseBatches");
const Courses_1 = require("../../../models/Courses");
const helpers_1 = require("../../../helpers");
let CourseAssignmentRelationsResolver = class CourseAssignmentRelationsResolver {
    async CourseBatches(courseAssignment, ctx) {
        return (0, helpers_1.getPrismaFromContext)(ctx).courseAssignment.findUnique({
            where: {
                id: courseAssignment.id,
            },
        }).CourseBatches({});
    }
    async courses(courseAssignment, ctx) {
        return (0, helpers_1.getPrismaFromContext)(ctx).courseAssignment.findUnique({
            where: {
                id: courseAssignment.id,
            },
        }).courses({});
    }
};
(0, tslib_1.__decorate)([
    TypeGraphQL.FieldResolver(_type => CourseBatches_1.CourseBatches, {
        nullable: false
    }),
    (0, tslib_1.__param)(0, TypeGraphQL.Root()),
    (0, tslib_1.__param)(1, TypeGraphQL.Ctx()),
    (0, tslib_1.__metadata)("design:type", Function),
    (0, tslib_1.__metadata)("design:paramtypes", [CourseAssignment_1.CourseAssignment, Object]),
    (0, tslib_1.__metadata)("design:returntype", Promise)
], CourseAssignmentRelationsResolver.prototype, "CourseBatches", null);
(0, tslib_1.__decorate)([
    TypeGraphQL.FieldResolver(_type => Courses_1.Courses, {
        nullable: false
    }),
    (0, tslib_1.__param)(0, TypeGraphQL.Root()),
    (0, tslib_1.__param)(1, TypeGraphQL.Ctx()),
    (0, tslib_1.__metadata)("design:type", Function),
    (0, tslib_1.__metadata)("design:paramtypes", [CourseAssignment_1.CourseAssignment, Object]),
    (0, tslib_1.__metadata)("design:returntype", Promise)
], CourseAssignmentRelationsResolver.prototype, "courses", null);
CourseAssignmentRelationsResolver = (0, tslib_1.__decorate)([
    TypeGraphQL.Resolver(_of => CourseAssignment_1.CourseAssignment)
], CourseAssignmentRelationsResolver);
exports.CourseAssignmentRelationsResolver = CourseAssignmentRelationsResolver;
