// @generated/type-graphql/enums/CoursesScalarFieldEnum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CoursesScalarFieldEnum = void 0;
const tslib_1 = require("tslib");
const TypeGraphQL = (0, tslib_1.__importStar)(require("type-graphql"));
var CoursesScalarFieldEnum;
(function (CoursesScalarFieldEnum) {
    CoursesScalarFieldEnum["id"] = "id";
    CoursesScalarFieldEnum["courseName"] = "courseName";
    CoursesScalarFieldEnum["courseDesc"] = "courseDesc";
    CoursesScalarFieldEnum["courseIntro"] = "courseIntro";
    CoursesScalarFieldEnum["instructorId"] = "instructorId";
    CoursesScalarFieldEnum["courseCategoryId"] = "courseCategoryId";
    CoursesScalarFieldEnum["createdAt"] = "createdAt";
    CoursesScalarFieldEnum["updateAt"] = "updateAt";
    CoursesScalarFieldEnum["courseStatus"] = "courseStatus";
    CoursesScalarFieldEnum["whatYouLearn"] = "whatYouLearn";
    CoursesScalarFieldEnum["coursePrice"] = "coursePrice";
})(CoursesScalarFieldEnum = exports.CoursesScalarFieldEnum || (exports.CoursesScalarFieldEnum = {}));
TypeGraphQL.registerEnumType(CoursesScalarFieldEnum, {
    name: "CoursesScalarFieldEnum",
    description: undefined,
});
