// @generated/type-graphql/enums/EnrollmentApprovalScalarFieldEnum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.EnrollmentApprovalScalarFieldEnum = void 0;
const tslib_1 = require("tslib");
const TypeGraphQL = (0, tslib_1.__importStar)(require("type-graphql"));
var EnrollmentApprovalScalarFieldEnum;
(function (EnrollmentApprovalScalarFieldEnum) {
    EnrollmentApprovalScalarFieldEnum["id"] = "id";
    EnrollmentApprovalScalarFieldEnum["studentName"] = "studentName";
    EnrollmentApprovalScalarFieldEnum["email"] = "email";
    EnrollmentApprovalScalarFieldEnum["course"] = "course";
    EnrollmentApprovalScalarFieldEnum["paymentMethod"] = "paymentMethod";
    EnrollmentApprovalScalarFieldEnum["amount"] = "amount";
    EnrollmentApprovalScalarFieldEnum["transactionId"] = "transactionId";
    EnrollmentApprovalScalarFieldEnum["status"] = "status";
})(EnrollmentApprovalScalarFieldEnum = exports.EnrollmentApprovalScalarFieldEnum || (exports.EnrollmentApprovalScalarFieldEnum = {}));
TypeGraphQL.registerEnumType(EnrollmentApprovalScalarFieldEnum, {
    name: "EnrollmentApprovalScalarFieldEnum",
    description: undefined,
});
