// @generated/type-graphql/enums/MyCourseScalarFieldEnum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MyCourseScalarFieldEnum = void 0;
const tslib_1 = require("tslib");
const TypeGraphQL = (0, tslib_1.__importStar)(require("type-graphql"));
var MyCourseScalarFieldEnum;
(function (MyCourseScalarFieldEnum) {
    MyCourseScalarFieldEnum["id"] = "id";
    MyCourseScalarFieldEnum["coursesId"] = "coursesId";
    MyCourseScalarFieldEnum["studentId"] = "studentId";
    MyCourseScalarFieldEnum["createdAt"] = "createdAt";
    MyCourseScalarFieldEnum["updateAt"] = "updateAt";
    MyCourseScalarFieldEnum["courseApproval"] = "courseApproval";
    MyCourseScalarFieldEnum["whyReject"] = "whyReject";
    MyCourseScalarFieldEnum["feeStatus"] = "feeStatus";
    MyCourseScalarFieldEnum["courseBatchesId"] = "courseBatchesId";
})(MyCourseScalarFieldEnum = exports.MyCourseScalarFieldEnum || (exports.MyCourseScalarFieldEnum = {}));
TypeGraphQL.registerEnumType(MyCourseScalarFieldEnum, {
    name: "MyCourseScalarFieldEnum",
    description: undefined,
});
