// @generated/type-graphql/enums/SuccessStoriesScalarFieldEnum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SuccessStoriesScalarFieldEnum = void 0;
const tslib_1 = require("tslib");
const TypeGraphQL = (0, tslib_1.__importStar)(require("type-graphql"));
var SuccessStoriesScalarFieldEnum;
(function (SuccessStoriesScalarFieldEnum) {
    SuccessStoriesScalarFieldEnum["id"] = "id";
    SuccessStoriesScalarFieldEnum["freelancingProfileUrl"] = "freelancingProfileUrl";
    SuccessStoriesScalarFieldEnum["paymentProof"] = "paymentProof";
    SuccessStoriesScalarFieldEnum["description"] = "description";
    SuccessStoriesScalarFieldEnum["status"] = "status";
    SuccessStoriesScalarFieldEnum["totalEarnedAmount"] = "totalEarnedAmount";
    SuccessStoriesScalarFieldEnum["city"] = "city";
    SuccessStoriesScalarFieldEnum["whyReject"] = "whyReject";
})(SuccessStoriesScalarFieldEnum = exports.SuccessStoriesScalarFieldEnum || (exports.SuccessStoriesScalarFieldEnum = {}));
TypeGraphQL.registerEnumType(SuccessStoriesScalarFieldEnum, {
    name: "SuccessStoriesScalarFieldEnum",
    description: undefined,
});
