// @generated/type-graphql/enums/UserScalarFieldEnum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserScalarFieldEnum = void 0;
const tslib_1 = require("tslib");
const TypeGraphQL = (0, tslib_1.__importStar)(require("type-graphql"));
var UserScalarFieldEnum;
(function (UserScalarFieldEnum) {
    UserScalarFieldEnum["id"] = "id";
    UserScalarFieldEnum["name"] = "name";
    UserScalarFieldEnum["email"] = "email";
    UserScalarFieldEnum["password"] = "password";
    UserScalarFieldEnum["cnic"] = "cnic";
    UserScalarFieldEnum["address"] = "address";
    UserScalarFieldEnum["phone"] = "phone";
    UserScalarFieldEnum["role"] = "role";
    UserScalarFieldEnum["permission"] = "permission";
    UserScalarFieldEnum["emailApproval"] = "emailApproval";
    UserScalarFieldEnum["successStoriesId"] = "successStoriesId";
    UserScalarFieldEnum["token"] = "token";
    UserScalarFieldEnum["createdAt"] = "createdAt";
    UserScalarFieldEnum["updateAt"] = "updateAt";
})(UserScalarFieldEnum = exports.UserScalarFieldEnum || (exports.UserScalarFieldEnum = {}));
TypeGraphQL.registerEnumType(UserScalarFieldEnum, {
    name: "UserScalarFieldEnum",
    description: undefined,
});
