// @generated/type-graphql/models/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.User = exports.SuccessStories = exports.Students = exports.Staff = exports.Speaker = exports.MyCourse = exports.MyCart = exports.Lectures = exports.Fees = exports.Faq = exports.Events = exports.EnrollmentApproval = exports.Courses = exports.CourseQuiz = exports.CourseBatches = exports.CourseAssignment = exports.ContactUs = exports.Comments = exports.Category = exports.Blogs = exports.Attendence = void 0;
var Attendence_1 = require("./Attendence");
Object.defineProperty(exports, "Attendence", { enumerable: true, get: function () { return Attendence_1.Attendence; } });
var Blogs_1 = require("./Blogs");
Object.defineProperty(exports, "Blogs", { enumerable: true, get: function () { return Blogs_1.Blogs; } });
var Category_1 = require("./Category");
Object.defineProperty(exports, "Category", { enumerable: true, get: function () { return Category_1.Category; } });
var Comments_1 = require("./Comments");
Object.defineProperty(exports, "Comments", { enumerable: true, get: function () { return Comments_1.Comments; } });
var ContactUs_1 = require("./ContactUs");
Object.defineProperty(exports, "ContactUs", { enumerable: true, get: function () { return ContactUs_1.ContactUs; } });
var CourseAssignment_1 = require("./CourseAssignment");
Object.defineProperty(exports, "CourseAssignment", { enumerable: true, get: function () { return CourseAssignment_1.CourseAssignment; } });
var CourseBatches_1 = require("./CourseBatches");
Object.defineProperty(exports, "CourseBatches", { enumerable: true, get: function () { return CourseBatches_1.CourseBatches; } });
var CourseQuiz_1 = require("./CourseQuiz");
Object.defineProperty(exports, "CourseQuiz", { enumerable: true, get: function () { return CourseQuiz_1.CourseQuiz; } });
var Courses_1 = require("./Courses");
Object.defineProperty(exports, "Courses", { enumerable: true, get: function () { return Courses_1.Courses; } });
var EnrollmentApproval_1 = require("./EnrollmentApproval");
Object.defineProperty(exports, "EnrollmentApproval", { enumerable: true, get: function () { return EnrollmentApproval_1.EnrollmentApproval; } });
var Events_1 = require("./Events");
Object.defineProperty(exports, "Events", { enumerable: true, get: function () { return Events_1.Events; } });
var Faq_1 = require("./Faq");
Object.defineProperty(exports, "Faq", { enumerable: true, get: function () { return Faq_1.Faq; } });
var Fees_1 = require("./Fees");
Object.defineProperty(exports, "Fees", { enumerable: true, get: function () { return Fees_1.Fees; } });
var Lectures_1 = require("./Lectures");
Object.defineProperty(exports, "Lectures", { enumerable: true, get: function () { return Lectures_1.Lectures; } });
var MyCart_1 = require("./MyCart");
Object.defineProperty(exports, "MyCart", { enumerable: true, get: function () { return MyCart_1.MyCart; } });
var MyCourse_1 = require("./MyCourse");
Object.defineProperty(exports, "MyCourse", { enumerable: true, get: function () { return MyCourse_1.MyCourse; } });
var Speaker_1 = require("./Speaker");
Object.defineProperty(exports, "Speaker", { enumerable: true, get: function () { return Speaker_1.Speaker; } });
var Staff_1 = require("./Staff");
Object.defineProperty(exports, "Staff", { enumerable: true, get: function () { return Staff_1.Staff; } });
var Students_1 = require("./Students");
Object.defineProperty(exports, "Students", { enumerable: true, get: function () { return Students_1.Students; } });
var SuccessStories_1 = require("./SuccessStories");
Object.defineProperty(exports, "SuccessStories", { enumerable: true, get: function () { return SuccessStories_1.SuccessStories; } });
var User_1 = require("./User");
Object.defineProperty(exports, "User", { enumerable: true, get: function () { return User_1.User; } });
