// @generated/type-graphql/resolvers/relations/User/args/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserMyCourseArgs = exports.UserMyCommentsArgs = exports.UserMyCartArgs = exports.UserFeesArgs = exports.UserCoursesArgs = exports.UserAttendenceArgs = void 0;
var UserAttendenceArgs_1 = require("./UserAttendenceArgs");
Object.defineProperty(exports, "UserAttendenceArgs", { enumerable: true, get: function () { return UserAttendenceArgs_1.UserAttendenceArgs; } });
var UserCoursesArgs_1 = require("./UserCoursesArgs");
Object.defineProperty(exports, "UserCoursesArgs", { enumerable: true, get: function () { return UserCoursesArgs_1.UserCoursesArgs; } });
var UserFeesArgs_1 = require("./UserFeesArgs");
Object.defineProperty(exports, "UserFeesArgs", { enumerable: true, get: function () { return UserFeesArgs_1.UserFeesArgs; } });
var UserMyCartArgs_1 = require("./UserMyCartArgs");
Object.defineProperty(exports, "UserMyCartArgs", { enumerable: true, get: function () { return UserMyCartArgs_1.UserMyCartArgs; } });
var UserMyCommentsArgs_1 = require("./UserMyCommentsArgs");
Object.defineProperty(exports, "UserMyCommentsArgs", { enumerable: true, get: function () { return UserMyCommentsArgs_1.UserMyCommentsArgs; } });
var UserMyCourseArgs_1 = require("./UserMyCourseArgs");
Object.defineProperty(exports, "UserMyCourseArgs", { enumerable: true, get: function () { return UserMyCourseArgs_1.UserMyCourseArgs; } });
