// @generated/type-graphql/resolvers/crud/Category/CategoryCrudResolver.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CategoryCrudResolver = void 0;
const tslib_1 = require("tslib");
const TypeGraphQL = (0, tslib_1.__importStar)(require("type-graphql"));
const graphql_fields_1 = (0, tslib_1.__importDefault)(require("graphql-fields"));
const AggregateCategoryArgs_1 = require("./args/AggregateCategoryArgs");
const CreateCategoryArgs_1 = require("./args/CreateCategoryArgs");
const CreateManyCategoryArgs_1 = require("./args/CreateManyCategoryArgs");
const DeleteCategoryArgs_1 = require("./args/DeleteCategoryArgs");
const DeleteManyCategoryArgs_1 = require("./args/DeleteManyCategoryArgs");
const FindFirstCategoryArgs_1 = require("./args/FindFirstCategoryArgs");
const FindManyCategoryArgs_1 = require("./args/FindManyCategoryArgs");
const FindUniqueCategoryArgs_1 = require("./args/FindUniqueCategoryArgs");
const GroupByCategoryArgs_1 = require("./args/GroupByCategoryArgs");
const UpdateCategoryArgs_1 = require("./args/UpdateCategoryArgs");
const UpdateManyCategoryArgs_1 = require("./args/UpdateManyCategoryArgs");
const UpsertCategoryArgs_1 = require("./args/UpsertCategoryArgs");
const helpers_1 = require("../../../helpers");
const Category_1 = require("../../../models/Category");
const AffectedRowsOutput_1 = require("../../outputs/AffectedRowsOutput");
const AggregateCategory_1 = require("../../outputs/AggregateCategory");
const CategoryGroupBy_1 = require("../../outputs/CategoryGroupBy");
let CategoryCrudResolver = class CategoryCrudResolver {
    async category(ctx, info, args) {
        const { _count } = (0, helpers_1.transformFields)((0, graphql_fields_1.default)(info));
        return (0, helpers_1.getPrismaFromContext)(ctx).category.findUnique({
            ...args,
            ...(_count && (0, helpers_1.transformCountFieldIntoSelectRelationsCount)(_count)),
        });
    }
    async findFirstCategory(ctx, info, args) {
        const { _count } = (0, helpers_1.transformFields)((0, graphql_fields_1.default)(info));
        return (0, helpers_1.getPrismaFromContext)(ctx).category.findFirst({
            ...args,
            ...(_count && (0, helpers_1.transformCountFieldIntoSelectRelationsCount)(_count)),
        });
    }
    async categories(ctx, info, args) {
        const { _count } = (0, helpers_1.transformFields)((0, graphql_fields_1.default)(info));
        return (0, helpers_1.getPrismaFromContext)(ctx).category.findMany({
            ...args,
            ...(_count && (0, helpers_1.transformCountFieldIntoSelectRelationsCount)(_count)),
        });
    }
    async createCategory(ctx, info, args) {
        const { _count } = (0, helpers_1.transformFields)((0, graphql_fields_1.default)(info));
        return (0, helpers_1.getPrismaFromContext)(ctx).category.create({
            ...args,
            ...(_count && (0, helpers_1.transformCountFieldIntoSelectRelationsCount)(_count)),
        });
    }
    async createManyCategory(ctx, info, args) {
        const { _count } = (0, helpers_1.transformFields)((0, graphql_fields_1.default)(info));
        return (0, helpers_1.getPrismaFromContext)(ctx).category.createMany({
            ...args,
            ...(_count && (0, helpers_1.transformCountFieldIntoSelectRelationsCount)(_count)),
        });
    }
    async deleteCategory(ctx, info, args) {
        const { _count } = (0, helpers_1.transformFields)((0, graphql_fields_1.default)(info));
        return (0, helpers_1.getPrismaFromContext)(ctx).category.delete({
            ...args,
            ...(_count && (0, helpers_1.transformCountFieldIntoSelectRelationsCount)(_count)),
        });
    }
    async updateCategory(ctx, info, args) {
        const { _count } = (0, helpers_1.transformFields)((0, graphql_fields_1.default)(info));
        return (0, helpers_1.getPrismaFromContext)(ctx).category.update({
            ...args,
            ...(_count && (0, helpers_1.transformCountFieldIntoSelectRelationsCount)(_count)),
        });
    }
    async deleteManyCategory(ctx, info, args) {
        const { _count } = (0, helpers_1.transformFields)((0, graphql_fields_1.default)(info));
        return (0, helpers_1.getPrismaFromContext)(ctx).category.deleteMany({
            ...args,
            ...(_count && (0, helpers_1.transformCountFieldIntoSelectRelationsCount)(_count)),
        });
    }
    async updateManyCategory(ctx, info, args) {
        const { _count } = (0, helpers_1.transformFields)((0, graphql_fields_1.default)(info));
        return (0, helpers_1.getPrismaFromContext)(ctx).category.updateMany({
            ...args,
            ...(_count && (0, helpers_1.transformCountFieldIntoSelectRelationsCount)(_count)),
        });
    }
    async upsertCategory(ctx, info, args) {
        const { _count } = (0, helpers_1.transformFields)((0, graphql_fields_1.default)(info));
        return (0, helpers_1.getPrismaFromContext)(ctx).category.upsert({
            ...args,
            ...(_count && (0, helpers_1.transformCountFieldIntoSelectRelationsCount)(_count)),
        });
    }
    async aggregateCategory(ctx, info, args) {
        return (0, helpers_1.getPrismaFromContext)(ctx).category.aggregate({
            ...args,
            ...(0, helpers_1.transformFields)((0, graphql_fields_1.default)(info)),
        });
    }
    async groupByCategory(ctx, info, args) {
        const { _count, _avg, _sum, _min, _max } = (0, helpers_1.transformFields)((0, graphql_fields_1.default)(info));
        return (0, helpers_1.getPrismaFromContext)(ctx).category.groupBy({
            ...args,
            ...Object.fromEntries(Object.entries({ _count, _avg, _sum, _min, _max }).filter(([_, v]) => v != null)),
        });
    }
};
(0, tslib_1.__decorate)([
    TypeGraphQL.Query(_returns => Category_1.Category, {
        nullable: true
    }),
    (0, tslib_1.__param)(0, TypeGraphQL.Ctx()),
    (0, tslib_1.__param)(1, TypeGraphQL.Info()),
    (0, tslib_1.__param)(2, TypeGraphQL.Args()),
    (0, tslib_1.__metadata)("design:type", Function),
    (0, tslib_1.__metadata)("design:paramtypes", [Object, Object, FindUniqueCategoryArgs_1.FindUniqueCategoryArgs]),
    (0, tslib_1.__metadata)("design:returntype", Promise)
], CategoryCrudResolver.prototype, "category", null);
(0, tslib_1.__decorate)([
    TypeGraphQL.Query(_returns => Category_1.Category, {
        nullable: true
    }),
    (0, tslib_1.__param)(0, TypeGraphQL.Ctx()),
    (0, tslib_1.__param)(1, TypeGraphQL.Info()),
    (0, tslib_1.__param)(2, TypeGraphQL.Args()),
    (0, tslib_1.__metadata)("design:type", Function),
    (0, tslib_1.__metadata)("design:paramtypes", [Object, Object, FindFirstCategoryArgs_1.FindFirstCategoryArgs]),
    (0, tslib_1.__metadata)("design:returntype", Promise)
], CategoryCrudResolver.prototype, "findFirstCategory", null);
(0, tslib_1.__decorate)([
    TypeGraphQL.Query(_returns => [Category_1.Category], {
        nullable: false
    }),
    (0, tslib_1.__param)(0, TypeGraphQL.Ctx()),
    (0, tslib_1.__param)(1, TypeGraphQL.Info()),
    (0, tslib_1.__param)(2, TypeGraphQL.Args()),
    (0, tslib_1.__metadata)("design:type", Function),
    (0, tslib_1.__metadata)("design:paramtypes", [Object, Object, FindManyCategoryArgs_1.FindManyCategoryArgs]),
    (0, tslib_1.__metadata)("design:returntype", Promise)
], CategoryCrudResolver.prototype, "categories", null);
(0, tslib_1.__decorate)([
    TypeGraphQL.Mutation(_returns => Category_1.Category, {
        nullable: false
    }),
    (0, tslib_1.__param)(0, TypeGraphQL.Ctx()),
    (0, tslib_1.__param)(1, TypeGraphQL.Info()),
    (0, tslib_1.__param)(2, TypeGraphQL.Args()),
    (0, tslib_1.__metadata)("design:type", Function),
    (0, tslib_1.__metadata)("design:paramtypes", [Object, Object, CreateCategoryArgs_1.CreateCategoryArgs]),
    (0, tslib_1.__metadata)("design:returntype", Promise)
], CategoryCrudResolver.prototype, "createCategory", null);
(0, tslib_1.__decorate)([
    TypeGraphQL.Mutation(_returns => AffectedRowsOutput_1.AffectedRowsOutput, {
        nullable: false
    }),
    (0, tslib_1.__param)(0, TypeGraphQL.Ctx()),
    (0, tslib_1.__param)(1, TypeGraphQL.Info()),
    (0, tslib_1.__param)(2, TypeGraphQL.Args()),
    (0, tslib_1.__metadata)("design:type", Function),
    (0, tslib_1.__metadata)("design:paramtypes", [Object, Object, CreateManyCategoryArgs_1.CreateManyCategoryArgs]),
    (0, tslib_1.__metadata)("design:returntype", Promise)
], CategoryCrudResolver.prototype, "createManyCategory", null);
(0, tslib_1.__decorate)([
    TypeGraphQL.Mutation(_returns => Category_1.Category, {
        nullable: true
    }),
    (0, tslib_1.__param)(0, TypeGraphQL.Ctx()),
    (0, tslib_1.__param)(1, TypeGraphQL.Info()),
    (0, tslib_1.__param)(2, TypeGraphQL.Args()),
    (0, tslib_1.__metadata)("design:type", Function),
    (0, tslib_1.__metadata)("design:paramtypes", [Object, Object, DeleteCategoryArgs_1.DeleteCategoryArgs]),
    (0, tslib_1.__metadata)("design:returntype", Promise)
], CategoryCrudResolver.prototype, "deleteCategory", null);
(0, tslib_1.__decorate)([
    TypeGraphQL.Mutation(_returns => Category_1.Category, {
        nullable: true
    }),
    (0, tslib_1.__param)(0, TypeGraphQL.Ctx()),
    (0, tslib_1.__param)(1, TypeGraphQL.Info()),
    (0, tslib_1.__param)(2, TypeGraphQL.Args()),
    (0, tslib_1.__metadata)("design:type", Function),
    (0, tslib_1.__metadata)("design:paramtypes", [Object, Object, UpdateCategoryArgs_1.UpdateCategoryArgs]),
    (0, tslib_1.__metadata)("design:returntype", Promise)
], CategoryCrudResolver.prototype, "updateCategory", null);
(0, tslib_1.__decorate)([
    TypeGraphQL.Mutation(_returns => AffectedRowsOutput_1.AffectedRowsOutput, {
        nullable: false
    }),
    (0, tslib_1.__param)(0, TypeGraphQL.Ctx()),
    (0, tslib_1.__param)(1, TypeGraphQL.Info()),
    (0, tslib_1.__param)(2, TypeGraphQL.Args()),
    (0, tslib_1.__metadata)("design:type", Function),
    (0, tslib_1.__metadata)("design:paramtypes", [Object, Object, DeleteManyCategoryArgs_1.DeleteManyCategoryArgs]),
    (0, tslib_1.__metadata)("design:returntype", Promise)
], CategoryCrudResolver.prototype, "deleteManyCategory", null);
(0, tslib_1.__decorate)([
    TypeGraphQL.Mutation(_returns => AffectedRowsOutput_1.AffectedRowsOutput, {
        nullable: false
    }),
    (0, tslib_1.__param)(0, TypeGraphQL.Ctx()),
    (0, tslib_1.__param)(1, TypeGraphQL.Info()),
    (0, tslib_1.__param)(2, TypeGraphQL.Args()),
    (0, tslib_1.__metadata)("design:type", Function),
    (0, tslib_1.__metadata)("design:paramtypes", [Object, Object, UpdateManyCategoryArgs_1.UpdateManyCategoryArgs]),
    (0, tslib_1.__metadata)("design:returntype", Promise)
], CategoryCrudResolver.prototype, "updateManyCategory", null);
(0, tslib_1.__decorate)([
    TypeGraphQL.Mutation(_returns => Category_1.Category, {
        nullable: false
    }),
    (0, tslib_1.__param)(0, TypeGraphQL.Ctx()),
    (0, tslib_1.__param)(1, TypeGraphQL.Info()),
    (0, tslib_1.__param)(2, TypeGraphQL.Args()),
    (0, tslib_1.__metadata)("design:type", Function),
    (0, tslib_1.__metadata)("design:paramtypes", [Object, Object, UpsertCategoryArgs_1.UpsertCategoryArgs]),
    (0, tslib_1.__metadata)("design:returntype", Promise)
], CategoryCrudResolver.prototype, "upsertCategory", null);
(0, tslib_1.__decorate)([
    TypeGraphQL.Query(_returns => AggregateCategory_1.AggregateCategory, {
        nullable: false
    }),
    (0, tslib_1.__param)(0, TypeGraphQL.Ctx()),
    (0, tslib_1.__param)(1, TypeGraphQL.Info()),
    (0, tslib_1.__param)(2, TypeGraphQL.Args()),
    (0, tslib_1.__metadata)("design:type", Function),
    (0, tslib_1.__metadata)("design:paramtypes", [Object, Object, AggregateCategoryArgs_1.AggregateCategoryArgs]),
    (0, tslib_1.__metadata)("design:returntype", Promise)
], CategoryCrudResolver.prototype, "aggregateCategory", null);
(0, tslib_1.__decorate)([
    TypeGraphQL.Query(_returns => [CategoryGroupBy_1.CategoryGroupBy], {
        nullable: false
    }),
    (0, tslib_1.__param)(0, TypeGraphQL.Ctx()),
    (0, tslib_1.__param)(1, TypeGraphQL.Info()),
    (0, tslib_1.__param)(2, TypeGraphQL.Args()),
    (0, tslib_1.__metadata)("design:type", Function),
    (0, tslib_1.__metadata)("design:paramtypes", [Object, Object, GroupByCategoryArgs_1.GroupByCategoryArgs]),
    (0, tslib_1.__metadata)("design:returntype", Promise)
], CategoryCrudResolver.prototype, "groupByCategory", null);
CategoryCrudResolver = (0, tslib_1.__decorate)([
    TypeGraphQL.Resolver(_of => Category_1.Category)
], CategoryCrudResolver);
exports.CategoryCrudResolver = CategoryCrudResolver;
