// typegraphql-prisma/lib/generator/resolvers/helpers.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.generateCrudResolverClassMethodDeclaration = void 0;
const ts_morph_1 = require("ts-morph");
const types_1 = require("../dmmf/types");
function generateCrudResolverClassMethodDeclaration(action, mapping, dmmfDocument) {
    return {
        name: action.name,
        isAsync: true,
        returnType: `Promise<${action.returnTSType}>`,
        decorators: [
            {
                name: `TypeGraphQL.${action.operation}`,
                arguments: [
                    `_returns => ${action.typeGraphQLType}`,
                    ts_morph_1.Writers.object({
                        nullable: `${!action.method.isRequired}`,
                    }),
                ],
            },
        ],
        parameters: [
            {
                name: "ctx",
                // TODO: import custom `ContextType`
                type: "any",
                decorators: [{ name: "TypeGraphQL.Ctx", arguments: [] }],
            },
            {
                name: "info",
                type: "GraphQLResolveInfo",
                decorators: [{ name: "TypeGraphQL.Info", arguments: [] }],
            },
            ...(!action.argsTypeName
                ? []
                : [
                    {
                        name: "args",
                        type: action.argsTypeName,
                        decorators: [{ name: "TypeGraphQL.Args", arguments: [] }],
                    },
                ]),
        ],
        statements: action.kind === types_1.DMMF.ModelAction.aggregate
            ? [
                /* ts */ ` return getPrismaFromContext(ctx).${mapping.collectionName}.${action.kind}({
              ...args,
              ...transformFields(graphqlFields(info as any)),
            });`,
            ]
            : action.kind === types_1.DMMF.ModelAction.groupBy
                ? [
                    /* ts */ ` const { _count, _avg, _sum, _min, _max } = transformFields(
              graphqlFields(info as any)
            );`,
                    /* ts */ ` return getPrismaFromContext(ctx).${mapping.collectionName}.${action.kind}({
              ...args,
              ...Object.fromEntries(
                Object.entries({ _count, _avg, _sum, _min, _max }).filter(([_, v]) => v != null)
              ),
            });`,
                ]
                : [
                    /* ts */ ` const { _count } = transformFields(
              graphqlFields(info as any)
            );
            return getPrismaFromContext(ctx).${mapping.collectionName}.${action.kind}({
              ...args,
              ...(_count && transformCountFieldIntoSelectRelationsCount(_count)),
            });`,
                ],
    };
}
exports.generateCrudResolverClassMethodDeclaration = generateCrudResolverClassMethodDeclaration;
